src/parser/mod.rs

```diff
@@ -10540,6 +10567,16 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_optional_idents(&mut self) -> Result<Option<Vec<Ident>>, ParserError> {
+        if self.consume_token(&Token::LParen) {
+            let partitions = self.parse_comma_separated(|p| p.parse_identifier(false))?;
+            self.expect_token(&Token::RParen)?;
+            Ok(Some(partitions))
+        } else {
+            Ok(None)
+        }
+    }
```

Comment:
instead of duplicating this function we have the caller do
```rust
let engine_parameters = if self.peek_token() == LParen {
   Some(self.parse_partitions())
} else {
    None
}
```
then makes sense to rename `parse_partitions` to something more generic now like `parse_parenthesized_identifiers` or similar
```

Reply:
Renamed `parse_partitions` -> `parse_parenthesized_identifiers ` and updated code.